docs(auth-kit): Update docs for the new `SafeAuthPack` (#273)
safe-global/safe-docs: safe-core-sdk/auth-kit/safe-auth.md

# Integration with SafeAuth

This guide demonstrate how to create an externally-owned account using your email or social media account. Once authenticated, you can sign transactions and interact with any Safe accounts you own.

The `SafeAuthPack` is an authentication system that utilizes the [Web3Auth](https://web3auth.io) MPC technology. It was developed in collaboration with Web3Auth to create a smooth onboarding experience for web2 users across different dApps.

## Prerequisites

- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

### Install dependencies

```bash
npm i @safe-global/auth-kit @web3auth/safeauth-embed
```

## Create a SafeAuthPack instance

> We are going to use the provided `SafeAuthPack` exported in the `@safe-global/auth-kit` package.

Reviewer: ```suggestion
We will use the provided `SafeAuthPack` exported in the `@safe-global/auth-kit` package.
```

  safes?: string[] // The list of associated Safe addresses in the chain
}
```

The `signOut()` method removes the current session.

```typescript
await safeAuthPack.signOut();
```

After the user is authenticated, call `getProvider()` to get the Ethereum provider instance. This is a [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compatible provider you can wrap using your favorite library (web3, ethers).

```typescript
safeAuthPack.getProvider();
```

We expose two methods for listening to events, `subscribe()` and `unsubscribe()`.

```typescript
const accountChangedHandler = (accounts: string[]) => {
  console.log('Signer accounts:', accounts);
};

safeAuthPack.subscribe('accountsChanged', accountChangedHandler);
safeAuthPack.unsubscribe('accountsChanged', accountChangedHandler);
```

The `SafeAuthPack` instantiation will return the list of associated Safe addresses as part of the response from the `signIn()` method when the `txServiceUrl` is provided.

```typescript
const safeAuthPack = new SafeAuthPack({
  txServiceUrl: 'https://safe-transaction-mainnet.safe.global',
});
```

## Signing and executing transactions using the SafeAuthPack and Protocol Kit

> The `SafeAuthPack` can be used in conjunction with the [Protocol Kit](../protocol-kit/README.md) to establish a connection to a Safe. This connection is made using the `provider` and `signer` associated with the authenticated account.

Reviewer: ```suggestion
The `SafeAuthPack` can be used with the [Protocol Kit](../protocol-kit/README.md) to establish a connection to a Safe. This connection is made using the `provider` and `signer` associated with the authenticated account.
```
